src/pages/word-duel.jsx

```diff
@@ -26,70 +19,63 @@ export default function WordDuel() {
 
     function checkGuess(guess) {
 
-        guess = guess.split('');
-
-        if (guess.length !== word.length) {
-            setMessage("Guess wrong length!");
+        if (util.isMaxGuesses(guesses)) {
+            setMessage("Maximum number of guesses already reached.");
+            setGameOver(true);
             return false;
         }
 
-        if (guesses.length >= MAX_GUESSES) {
-            setMessage("Guess limit reached.");
-            setGameOver(true);
+        if (!util.isWordLength(word, guess)) {
+            setMessage("Guess length does not match word length");
             return false;
         }
 
         setMessage("");
-        setGuesses([...guesses, guess]);
         updateActive("");
-        evaluateGuess(guess);
+
+        setGuesses([...guesses, guess]);
+
+        const evaluation = util.getEvaluation(word, guess);
+        setEvaluations([...evaluations, evaluation]);
+
         return true;
     }
 
-    function evaluateGuess(guess) {
-        const evaluation = new Array(word.length).fill(EvaluationColor.Gray);
-        let wordCopy = word;
+    function checkWin() {
 
-        for (let i = 0; i < word.length; i++) {
-            if (guess[i] === word[i]) {
-                evaluation[i] = EvaluationColor.Green;
-                // make sure to replace with an impossible symbol
-                delete wordCopy[i];
-            }
-        }
-        
-        console.log({evaluation, guess, wordCopy});
-        
-        for (let i = 0; i < wordCopy.length; i++) {
-            const indexOfLetter = wordCopy.indexOf(guess[i]);
-            if (indexOfLetter > -1) {
-                evaluation[i] = EvaluationColor.Yellow;
-                // make sure to replace with an impossible symbol
-                delete wordCopy[indexOfLetter];
-            }
+        if (evaluations.length < 1) {
+            return;
         }
 
-        winGame(evaluation);
-        setEvaluations([...evaluations, evaluation]);
-    }
+        const evaluation = evaluations[evaluations.length - 1];
 
-    function winGame(evaluation) {
-        if (evaluation.some(result => result !== "green")) {
+        if (!util.isWin(evaluation)) {
```

Comment:
We can swap around the conditions here and it'll flow a little smoother

```javascript
if (util.isWin(evaluation)) {
    setGameOver(true);
    setMessage("You win!");
} else if (util.isMaxGuesses(guesses)) {
    setGameOver(true);
    setMessage("You lose!");
}
```

It might need refactoring later when you decide on an "end state" component, but you could do like:

```javascript
function endGame(message) {
    setGameOver(true);
    setMessage(message);
}

// ... and then

if (util.isWin(evaluation)) {
    endGame("You win!");
}
```